### PR TITLE
Enable -preview=fixAliasThis switch by default

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

**Comment by RazvanN7** [comment]
This needs a deprecation for the switch, I'm just testing to see what the autotester says.

File: changelog/dmd.preview-alias-this.dd [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Alias this fix has now become the default
+
+Prior to this release there was an alias this bug where
+[alias this was not considered when doing scope](https://issues.dlang.org/show_bug.cgi?id=16082).
```
Comment (dkorpel) [inline changelog/dmd.preview-alias-this.dd:4]: What is "doing scope"?

**Comment by WalterBright** [comment]
I thought we were going to leave `alias this` for classes alone and not try to fix it?

**Comment by RazvanN7** [comment]
The code is already there, but it's behind a preview switch. I thought we might as well keep, but the alternative would be to deprecate the switch and eventually delete this code. That works for me also.

**Comment by RazvanN7** [comment]
@WalterBright This PR affects both structs and classes.

File: changelog/dmd.preview-alias-this.dd (hunk 1 above)
Comment (thewilsonator) [reply changelog/dmd.preview-alias-this.dd:4]: I presume inferring scope

Comment (ibuclaw) [reply changelog/dmd.preview-alias-this.dd:4]: "when doing scope lookups"

**Pull request closed without merging**